label_studio/projects/models.py

```diff
@@ -592,10 +592,15 @@ def get_updated_weights(self):
             control_type = outputs[control_name]['type']
             if control_type in exclude_control_types:
                 continue
+
+            def get_label(label):
+                return self.control_weights.get(control_name, {}).get('labels', {}).get(label)
+
             control_weights[control_name] = {
                 'overall': self.control_weights.get(control_name, {}).get('overall') or 1.0,
                 'type': control_type,
-                'labels': {label: self.control_weights.get(control_name, {}).get('labels', {}).get(label) or 1.0 for label in outputs[control_name].get('labels', [])},
+                'labels': {label: get_label(label) if get_label(label) is not None else 1.0 for label in
```

Comment:
Would this call `get_label` twice per label? if get_label(label) is not None? Once at the conditional check `if get_label(label)`, then again when you assign the value to the key in the labels dict?